pvlib/data/gold/sdm.py

```diff
@@ -0,0 +1,832 @@
+        worst_i_test_terminal_max_abs_res = 0.
+        worst_v_test_current_sum_max_abs_res = 0.
+        worst_v_test_terminal_max_abs_res = 0.
+
+        # Initialize the worst results and benchmarks for this device
+        # Note that a caller may choose not to test all the possible functions
+        # FUTURE This dictionary can be expanded in a backwards compatible
+        #  manner to support additional gold values and residuals for
+        #  Isc, R@Isc, Ix, Pmp, Ixx, R@Voc, and Voc
+        device_dict = \
+            {"name": device["name"],
+             "Technology": device["Technology"],
+             "N_s": device["N_s"]}
+        if "i_from_v" in test_func:
+            device_dict["i_from_v"] = {
+                "current_sum": None,
+                "terminal": None,
+                "performance": {
+                    "time_s": {
+                        "data": [],
+                        "min": None,
+                        "med": None,
+                        "max": None
+                    }
+                }
+            }
+        if "v_from_i" in test_func:
+            device_dict["v_from_i"] = {
+                "current_sum": None,
+                "terminal": None,
+                "performance": {
+                    "time_s": {
+                        "data": [],
+                        "min": None,
+                        "med": None,
+                        "max": None
+                    }
+                }
+            }
+
+        # Gauge each I-V curve for current sum and terminal residuals
+        for iv_curve in device["iv_curves"]:
+            # Check if caller wants to test an i_from_v function
+            if "i_from_v" in test_func:
+                # Compute test values using provided test function that meets
+                #  pvlib API
+                start = timer()
+                i_test = test_func["i_from_v"](
+                    iv_curve["r_sh"], iv_curve["r_s"], iv_curve["nNsVth"],
+                    iv_curve["v_gold"], iv_curve["i_0"], iv_curve["i_l"])
+                device_dict["i_from_v"]["performance"]["time_s"][
+                    "data"].append(timer() - start)
+
+                # Gauge computed values against gold values for sum of
+                #  currents residual at diode node
+                i_test_current_sum_res = sum_current(
+                    iv_curve["r_sh"], iv_curve["r_s"], iv_curve["nNsVth"],
+                    i_test, iv_curve["v_gold"], iv_curve["i_0"],
+                    iv_curve["i_l"])
+                i_test_current_sum_max_abs_res = numpy.amax(
+                    numpy.abs(i_test_current_sum_res))
+                if worst_i_test_current_sum_max_abs_res < \
+                   i_test_current_sum_max_abs_res:
+                    worst_i_test_current_sum_max_abs_res = \
+                        i_test_current_sum_max_abs_res
+                    device_dict["i_from_v"]["current_sum"] = \
```

Comment:
These ``device_dict`` assignments look similar enough to refactor into a function and improve readability. 

```python
def _create_device_result(iv_curve, test_name, test_result, residual, max_abs_residual):
    out = {
        "G": iv_curve["G"],
        "T": iv_curve["T"],
        "i_l": iv_curve["i_l"],
        "i_0": iv_curve["i_0"],
        "r_s": iv_curve["r_s"],
        "r_sh": iv_curve["r_sh"],
        "nNsVth": iv_curve["nNsVth"],
        "v_gold": iv_curve["v_gold"].tolist(),
        "i_gold": iv_curve["i_gold"].tolist(),
        test_name: test_result.tolist(),
        "residual": residual.tolist(),
        "max_abs_residual": max_abs_residual.tolist(),
    }
    return out

device_dict["i_from_v"]["current_sum"] = _create_device_result(iv_curve, "i_test", i_test, i_test_current_sum_res, i_test_current_sum_max_abs_res)
```